pkg/app/piped/controller/planner.go

```diff
@@ -187,9 +188,16 @@ func (p *planner) Run(ctx context.Context) error {
 		)
 	}
 
+	deploymentStatus := p.doneDeploymentStatus
+
+	defer func() {
+		p.doneDeploymentStatus = deploymentStatus
+		controllermetrics.UpdateDeploymentStatus(p.deployment.Id, deploymentStatus, p.deployment.Kind, p.deployment.PlatformProvider)
+	}()
```

Comment:
Not sure why we need to use this `deploymentStatus` variable. Why not just add the defer function here which call to
```golang
	defer func() {
		controllermetrics.UpdateDeploymentStatus(p.deployment.Id, p.doneDeploymentStatus, p.deployment.Kind, p.deployment.PlatformProvider)
	}()
```

Reply:
Can't agree more :)
Let me fix it.